internal/controllers/objectdeployments/objectdeployment_controller_test.go

```diff
@@ -0,0 +1,179 @@
+package objectdeployments
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/api/errors"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	corev1alpha1 "package-operator.run/apis/core/v1alpha1"
+	"package-operator.run/internal/testutil"
+)
+
+func init() {
+	if err := corev1alpha1.AddToScheme(testScheme); err != nil {
+		panic(err)
+	}
+}
+
+func TestObjectDeploymentReconciler_Err(t *testing.T) {
+	t.Parallel()
+
+	fooErr := errors.NewBadRequest("foo")
+
+	clientMock := testutil.NewClient()
+	c := NewObjectDeploymentController(
+		clientMock, ctrl.Log.WithName("object deployment test"), testScheme)
+
+	clientMock.
+		On("Get", mock.Anything, mock.Anything, mock.AnythingOfType("*v1alpha1.ObjectDeployment"), mock.Anything).
+		Return(fooErr)
+
+	ctx := context.Background()
+	res, err := c.Reconcile(ctx, ctrl.Request{})
+
+	require.Error(t, err)
+	require.EqualError(t, err, fooErr.Error())
+	assert.Empty(t, res)
+	assert.True(t, res.IsZero())
+
+	clientMock.AssertExpectations(t)
+}
+
+func TestObjectDeploymentReconciler_NotFound(t *testing.T) {
+	t.Parallel()
+
+	fooErr := errors.NewBadRequest("foo")
```

Comment:
This test doesn't do what its name suggests. It's basically the same as the previous one. The point of this case should be to ensure that when the controller receives a "not found" error from `client.Get()`, it assumes that the object was deleted and, therefore, does not return an error. Instead of the `fooErr` use a `notFoundErr`. The `c.Reconcile()` call should then succeed.

It might be also nice to add an assertion that the "Update" method on the client wasn't called (because the object was deleted).

Reply:
Hi @Ankit152,
I realized that the explanation I gave might be a little confusing. Basically, we're trying to get here: [#L121](https://github.com/package-operator/package-operator/blob/main/internal/controllers/objectdeployments/objectdeployment_controller.go#L121), so that the `Reconcile` function returns a `nil` error. Please let me know, if you have any questions. We can even arrange a call, if you'd like.

Reply:
Let me have a look at it.

Reply:
Hey I have fixed the tests. Mistakenly, I wrote two tests for `BadRequest`.